zerver/models.py

```diff
@@ -2509,6 +2509,19 @@ def __str__(self) -> str:
                                                self.address or list(self.users.all()),
                                                self.scheduled_timestamp)
 
+class MissedMessageEmailAddress(models.Model):
+    EXPIRY_SECONDS = 60 * 60 * 24 * 5
+
+    message = models.ForeignKey(Message, on_delete=CASCADE)  # type: Message
+    user_profile = models.ForeignKey(UserProfile, on_delete=CASCADE)  # type: UserProfile
+    email_token = models.CharField(max_length=34, unique=True, db_index=True)  # type: str
+
+    timestamp = models.DateTimeField(db_index=True, default=timezone_now)  # type: datetime.datetime
+    used = models.BooleanField(default=False, db_index=True)  # type: bool
```

Comment:
Hmm, do you think an integer field `times_used` here (initially 0) would be better?  From a product perspective, I can imagine us changing the feature to allow e.g. up to 5-10 sends to one of these addresses... it certainly seems more correct to allow a limited number of replies from one's email client.

Reply:
Changed to this model. (details in a comment further down)